storage_tiering.cpp

```diff
@@ -763,7 +763,7 @@ namespace irods {
         std::string coll_name = p.parent_path().string();
 
         std::string qstr{boost::str(
-            boost::format("SELECT META_DATA_ATTR_VALUE WHERE DATA_NAME = '%s' AND COLL_NAME = '%s' AND META_DATA_ATTR_NAME = '%s' AND META_DATA_ATTR_UNITS = '%s'")
+            boost::format("SELECT META_DATA_ATTR_VALUE WHERE DATA_NAME = '%s' AND COLL_NAME = '%s' AND META_DATA_ATTR_NAME = '%s' AND META_DATA_ATTR_UNITS >= '%s'")
```

Comment:
I think we should just remove the `META_DATA_ATTR_UNITS` filter, as you suggested here: https://github.com/irods/irods_capability_storage_tiering/issues/232#issuecomment-1851900106
```suggestion
            boost::format("SELECT META_DATA_ATTR_VALUE WHERE DATA_NAME = '%s' AND COLL_NAME = '%s' AND META_DATA_ATTR_NAME = '%s'")
```

Reply:
Hi,
There are two ways to implement this:
* only change the sql query (very easy, but not complete)
* change the function `storage_tiering::get_group_name_by_replica_number` to not have the parameter `_replica_number` and change the sql query. This change means you have to fix all calls to the function.

Reply:
I believe we never want to depend on the specifics of replica numbers being in a particular layout / usage pattern (assuming replica `0`, or that one is greater/less than another, etc.).

So I think we must go with option B (and with it, lose the possibility of a data object being in multiple tier groups at the same time).

Discussion is still open of course.

Reply:
Right, replica numbers only go up so this seems like it would work. However, it makes me a little bit nervous for the reason @trel laid out, and doesn't really do anything more than just removing the replica number filter for our existing supported use cases.

Fortunately, I see only 2 uses of this function, so replacing it should be easy. I think your suggestion of renaming the function or making a new function called `get_group_name_for_object` or something along those lines is a good one.

Reply:
I am OK with option 2
If everybody agrees I will implement it somewhere next week. Just let me know.

Reply:
Agreed. Thanks!

Reply:
Added a new commit which renames `get_group_name_by_resource_number` to `get_group_name_for_object`.
Excluded ` AND META_DATA_ATTR_UNITS = '%s'` from sql query.
Removed parameter `_replica_number` from the function.

Tested locally OK.

Reply:
Excellent. Resolving